File: frontend/src/types/role.ts
export type Role = "None" | "Owner" | "Admin" | "Member" | "Guest";

export const ROLE_LABELS: Record<Role, string> = {
  None : "None",
  Owner: "Owner",
  Admin: "Admin",
  Member: "Member",
  Guest: "Guest",
}

export function getRoleLabel(role: Role) {
  return ROLE_LABELS[role] ?? role;
}

File: server/Domain/Enums/Workspace/Role.cs
using System.Text.Json.Serialization;

namespace Domain;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum Role
{
    Guest = 1,
    Member = 2,
    Admin = 3,
    Owner = 4,
}

public static class RoleExtensions
{
    public static bool IsAtLeast(this Role role, Role requiredRole)
    {
        return role >= requiredRole;
    }
}

